projects/batfish/src/test/java/org/batfish/main/BatfishTest.java

```diff
@@ -81,6 +86,26 @@ public void testAnswerBadQuestion() throws IOException {
   }
 
   @Test
+  public void testMultipleBestVrrpCandidates() throws IOException {
+    SortedMap<String, String> configurationsText = new TreeMap<>();
+    String[] configurationNames = new String[] {"r1", "r2"};
+    Ip vrrpAddress = new Ip("1.0.0.10");
+    String testConfigsPrefix = "org/batfish/grammar/cisco/testrigs/vrrp_multiple_best/configs/";
+    for (String configurationName : configurationNames) {
+      String configurationText = CommonUtil.readResource(testConfigsPrefix + configurationName);
+      configurationsText.put(configurationName, configurationText);
+    }
+    Batfish batfish = BatfishTestUtils.getBatfishFromConfigurationText(configurationsText, _folder);
+    SortedMap<String, Configuration> configurations = batfish.loadConfigurations();
+    Map<Ip, Set<String>> ipOwners = batfish.computeIpOwners(configurations, true);
+    Set<String> actualVrrpAddressOwners = ipOwners.get(vrrpAddress);
+    int expectedVrrpAddressOwnersSize = 1;
+    int actualVrrpAddressOwnersSize = actualVrrpAddressOwners.size();
+    assertTrue(actualVrrpAddressOwners.contains("r1"));
+    assertThat(expectedVrrpAddressOwnersSize, equalTo(actualVrrpAddressOwnersSize));
```

Comment:
I think you can simplify this and get better error messages:

assertThat(actualVrrpAddressOwners, contains("r1"));
assertThat(actualVrrpAddressOwners, hasSize(1));

or probably just

assertThat(ipOwners.get(vrrpAddress), equalTo(Collections.singleton("r1"));

Let the matchers do the work for you, because they have really nice toString functions.

Reply:
Used the latter. Fixed.